### PR TITLE
Add “Compile GPU kernels using ClangIR” Blog Post (GSoC 2024)

### PR DESCRIPTION
**Pull request by seven-mile**
@asl This is my write up for my GSoC 2024 project. I have tested this locally on Hugo. 😉

### CODE REVIEW

**Comment by asl** [comment]
@bcardosolopes Will you please review content-wise?

File: content/posts/2024-08-29-gsoc-opencl-c-support-for-clangir.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+7. **User Experience Enhancements**: Finally, we ensured that the end-to-end kernel compilation experience using ClangIR was smooth and intuitive, with minimal manual intervention required.
+
+# Results
+
+The project successfully met its primary goals. OpenCL kernels from the Polybench-GPU benchmark suite can now be compiled using ClangIR into LLVM IR for SPIR-V. All patches have been merged into the main ClangIR repository, and the project’s progress has been well-documented in the [overview issue](https://github.com/llvm/clangir/issues/689). I believe the work not only advanced OpenCL support but also laid a solid foundation for future enhancements, such as SYCL and CUDA support in ClangIR.
+
+We have successfully compiled and passed all 20 OpenCL C test cases from the [polybenchGpu](https://github.com/sgrauerg/polybenchGpu) repository. Please refer to our [artifact evaluation repository](https://github.com/seven-mile/clangir-ocl-ae) for detailed instructions on how to experiment with our work.
```
Comment (jopperm) [inline content/posts/2024-08-29-gsoc-opencl-c-support-for-clangir.md:36]:
```suggestion
We have successfully compiled and executed all 20 OpenCL C benchmarks from the [polybenchGpu](https://github.com/sgrauerg/polybenchGpu) repository, passing the built-in result validation. Please refer to our [artifact evaluation repository](https://github.com/seven-mile/clangir-ocl-ae) for detailed instructions on how to experiment with our work.
```
"pass" is a bit unspecific, you should highlight that you can actually run the kernels!

**Review by jopperm — APPROVED** [review]
Nice overview, thanks!

**Review by bcardosolopes — APPROVED** [review]
LGTM! Great work

**Comment by seven-mile** [comment]
Updated. : )

File: content/posts/2024-08-29-gsoc-opencl-c-support-for-clangir.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+---
+author: "Zhi Ma (7mile)"
+date: "2024-08-29"
```
Comment (asl) [inline content/posts/2024-08-29-gsoc-opencl-c-support-for-clangir.md:3]:
```suggestion
date: "2024-09-09"
```

**Review by asl — APPROVED** [review]
Thanks. Scheduled for next Monday

File: content/posts/2024-08-29-gsoc-opencl-c-support-for-clangir.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+# Background
+
+[The ClangIR project](https://github.com/llvm/clangir) aims to establish a new IR for Clang, built on top of MLIR. As part of the ongoing effort to support heterogeneous programming models, this project focuses on integrating OpenCL C language support into ClangIR. The ultimate goal is to enable the compilation of GPU kernels written in OpenCL C into LLVM IR targeting the SPIR-V architecture, laying the groundwork for future enhancements in SYCL and CUDA support.
+
+# What We Did
+
+Our work involved several key areas:
+
+1. **Address Space Support**: One of the fundamental tasks was teaching ClangIR to handle address spaces, a vital feature for languages like OpenCL. Initially, we considered mimicking LLVM's approach, but this proved inadequate for ClangIR's goals. After thorough discussion and an RFC, we implemented a unified address space design that aligns with ClangIR's objectives, ensuring a clean and maintainable code structure.
```
Comment (Lancern) [inline content/posts/2024-08-29-gsoc-opencl-c-support-for-clangir.md:18]: > After thorough discussion and an RFC

Maybe you could add a link to the mentioned RFC?

**Review by Lancern — COMMENTED** [review]
Congratulations on finishing your GSoC!

**Pull request merged**